execution/executor/src/lib.rs

```diff
@@ -516,14 +517,27 @@ where
             }
             outputs
         } else {
+            reexecuted_transactions = true;
             V::execute_block(transactions.clone(), &state_view)?
         };
 
         // Since other validators have committed these transactions, their status should all be
         // TransactionStatus::Keep.
-        for output in &vm_outputs {
-            if let TransactionStatus::Discard(_) = output.status() {
-                bail!("Syncing transactions that should be discarded.");
+        for (index, output) in vm_outputs.iter().enumerate() {
+            if let TransactionStatus::Discard(status_code) = output.status() {
+                let bail_error_message = format!(
+                    "Transaction version: {:?}\n
+                    Status code: {:?}\n
+                    Syncing transactions that have already been committed.",
```

Comment:
There's nothing wrong with syncing transactions that have already been committed (in fact, this is exactly what state sync is supposed to do). The issue here is that the transaction output says to discard the transaction, which *should never* happen (because if it was committed, it means it wasn't discarded 😄). 

Can we change this message to:
`Syncing a transaction that should be discarded! Transaction version: {:?}, status code: {:?}.`